src/main/groovy/com/github/winplay02/gitcraft/pipeline/Resetter.java

```diff
@@ -15,90 +17,66 @@
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.revwalk.RevWalk;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.List;
-import java.util.NavigableSet;
-import java.util.Objects;
-
-public class ResetStep extends Step {
-	@Override
-	public String getName() {
-		return STEP_RESET;
-	}
+import com.github.winplay02.gitcraft.GitCraft;
+import com.github.winplay02.gitcraft.types.OrderedVersion;
+import com.github.winplay02.gitcraft.util.MiscHelper;
+import com.github.winplay02.gitcraft.util.RepoWrapper;
 
-	@Override
-	public boolean preconditionsShouldRun(PipelineCache pipelineCache, OrderedVersion mcVersion, MappingFlavour mappingFlavour, MinecraftVersionGraph versionGraph, RepoWrapper repo) {
-		return GitCraft.config.refreshDecompilation && GitCraft.resetVersionGraph.getRootVersions().stream().map(mcVersion::compareTo).max(Comparator.naturalOrder()).orElseThrow() >= 0;
```

Comment:
Some preconditions (like config flags) should still be checked, at least in the run method

Reply:
I thought I covered them all, but I'll look through it again to see if I missed anything.